hasjob/views/index.py

```diff
@@ -338,13 +338,17 @@ def index(basequery=None, type=None, category=None, md5sum=None, domain=None,
     elif pinsandposts:
         g.impressions = {post.id: (pinflag, post.id, is_bgroup) for pinflag, post, is_bgroup in pinsandposts}
 
+    query_params = dict(request.args)
```

Comment:
The reason this threw me off is that in Python, `dict({...})` always returns a shallow copy:

```python
>>> a = {1: 2}
>>> b = dict(a)
>>> a == b
True
>>> a is b
False
>>> 
```

But in this case with `request.args` being a dict-like object, `request.args['key']` always being a single value in the original object but becoming a list in the copied object is inconsistent behaviour. This is a flaw in Werkzeug's design and something they likely cannot fix now. I'd therefore steer away from using `dict(...)` and use the `to_dict(...)` call instead, just to avoid this confusion.